### DISCUSSION TITLE
how do I mock error conditions in unit tests?

### DISCUSSION THREAD

**Discussion by smousa** [redpanda-data/connect#2443] [Q&A] [answered]
Follow up question based on: https://github.com/benthosdev/benthos/discussions/2442#discussion-6420195

Is there a way that I can simulate errors with the test framework for existing components, so I can make sure that the pipeline works as expected?

**Comment by mihaitodor** [comment, marked as answer]
Hey @smousa 👋 Sure! You can use [mocking](https://www.benthos.dev/docs/configuration/unit_testing#mocking-processors) and have `mapping: 'root = throw("your specific error")'`. For example, you could do `mapping: 'root = throw("key does not exist")'` to simulate a cache miss.

**Reply by smousa** [reply]
Oh, I couldn't find that in the docs, where would I expect to find that?

**Reply by mihaitodor** [reply]
Do you mean [`throw()`](https://www.benthos.dev/docs/guides/bloblang/functions#throw)? I know the search feature has some issues and, right now, bloblang methods don't show up in results for some reason... The functions seem to be indexed correctly though.
